src/main/java/software/amazon/kinesis/connectors/flink/internals/publisher/fanout/FanOutRecordPublisher.java

```diff
@@ -150,12 +150,16 @@ private RecordPublisherRunResult runWithBackoff(
 				return RecordPublisherRunResult.COMPLETE;
 			}
 
-			if (attempt == configuration.getSubscribeToShardMaxRetries()) {
-				throw new RuntimeException("Maximum reties exceeded for SubscribeToShard. " +
-					"Failed " + configuration.getSubscribeToShardMaxRetries() + " times.");
+			if (!(ex.getCause() instanceof io.netty.handler.timeout.ReadTimeoutException)) {
```

Comment:
With this approach ReadTimeoutExceptions would be retried indefinitely. Is that an expected behaviour?

Reply:
As discussed, yes. In high backpressure scenarios consumer will fail gracefully and reacquire subscription once the processing catches up. This would continue indefinitely, user would see `millisBehindLatest` increasing resulting in nudge for scaling/processing optimisation

Reply:
As discussed we will make the exception thrown from FanOutShardSubscriber explicitly transient to avoid making this decision based on ReadTimeoutException type.

Reply:
Done